assets/js/modules/analytics-4/utils/report-rows-with-set-values.test.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Analytics-4 filter rows having (not set) values tests.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { reportRowsWithSetValues } from './report-rows-with-set-values';
+
+describe( 'reportRowsWithSetValues', () => {
+	const mockRows = [
+		{ dimensionValues: [ { value: 'value1' } ] },
+		{ dimensionValues: [ { value: '(not set)' } ] },
+		{ dimensionValues: [ { value: 'value2' } ] },
+		{ dimensionValues: [ { value: 'value3' } ] },
+		{ dimensionValues: [ { value: 'value4' } ] },
+	];
```

Comment:
Let's define this in a `beforeEach()` to make the tests more robust. As things stand, if we consider `reportRowsWithSetValues()` as a black box, it could in theory mutate this data structure between tests.

```suggestion
	let mockRows;

	beforeEach( () => {
		mockRows = [
			{ dimensionValues: [ { value: 'value1' } ] },
			{ dimensionValues: [ { value: '(not set)' } ] },
			{ dimensionValues: [ { value: 'value2' } ] },
			{ dimensionValues: [ { value: 'value3' } ] },
			{ dimensionValues: [ { value: 'value4' } ] },
		];
	} );
```